build/bin/install_tarball

```diff
@@ -2,59 +2,63 @@
-${INSTALL_COMMAND} -m 0755 queries-postgres-uptime.yml ${PMM_DIR}/collectors/custom-queries/postgresql/high-resolution
-${INSTALL_COMMAND} -m 0755 queries.yaml ${PMM_DIR}/collectors/custom-queries/postgresql/medium-resolution
-${INSTALL_COMMAND} -m 0755 queries-mysqld-group-replication.yml ${PMM_DIR}/collectors/custom-queries/mysql/high-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/bin
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/tools
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/exporters
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/config
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/textfile-collector
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/textfile-collector/low-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/textfile-collector/medium-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/textfile-collector/high-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/mysql
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/mysql/low-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/mysql/medium-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/mysql/high-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/postgresql
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/postgresql/low-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/postgresql/medium-resolution
+${INSTALL_COMMAND} -m 0755 -d "${PMM_DIR}"/collectors/custom-queries/postgresql/high-resolution
+${INSTALL_COMMAND} -m 0755 queries-postgres-uptime.yml "${PMM_DIR}"/collectors/custom-queries/postgresql/high-resolution
+${INSTALL_COMMAND} -m 0755 queries.yaml "${PMM_DIR}"/collectors/custom-queries/postgresql/medium-resolution
+${INSTALL_COMMAND} -m 0755 queries-mysqld-group-replication.yml "${PMM_DIR}"/collectors/custom-queries/mysql/high-resolution
 
-for FILE in $( ls ${PWD}/bin ); do
-    if [ "x${FILE}" = "xpmm-admin" -o "x${FILE}" = "xpmm-agent" ]; then
-        ${INSTALL_COMMAND} -m 0755 ${PWD}/bin/${FILE} ${PMM_DIR}/bin
-    elif [ "x${FILE}" = "xpt-summary" -o "x${FILE}" = "xpt-mysql-summary" -o "x${FILE}" = "xpt-pg-summary" -o "x${FILE}" = "xpt-mongodb-summary" ]; then
-        ${INSTALL_COMMAND} -m 0755 ${PWD}/bin/${FILE} ${PMM_DIR}/tools
+find ./bin -maxdepth 1 -type f | while read -r FILE; do
+    if [ "x${FILE}" = "xpmm-admin" ] || [ "x${FILE}" = "xpmm-agent" ]; then
+        ${INSTALL_COMMAND} -m 0755 "${FILE}" "${PMM_DIR}"/bin
+    elif [ "x${FILE}" = "xpt-summary" ] || [ "x${FILE}" = "xpt-mysql-summary" ] || [ "x${FILE}" = "xpt-pg-summary" ] || [ "x${FILE}" = "xpt-mongodb-summary" ]; then
+        ${INSTALL_COMMAND} -m 0755 "${FILE}" "${PMM_DIR}"/tools
```

Comment:
To be honest, readability is worse than was.

In my opinion, we should write scripts as clearly as possible and this should be a priority.

Let's change it to:
```
        ${INSTALL_COMMAND} -m 0755 "${FILE}" "${PMM_DIR}/tools"
```
Here and in other places.

Reply:
@nikita-b 

```bash
[ "x${FILE}" = "xpmm-admin" ] || [ "x${FILE}" = "xpmm-agent" ]
```

I would argue that boolean operations are more readable as the syntax is commonly present in other languages, plus it is better defined in shells.

Similarly,

```text
"${PMM_DIR}/tools"
```

.. this is less readable, especially where there is no syntax-highlighting, to pick out the variable as you then need to read backwards from the `"`, which draws the eye to the end. The prevention of globbing is specific to the user-defined values, so where does that leave the other parts of the script where it is not necessary?

For example:
```bash
${RESOLUTION}-resolution
${INSTALL_COMMAND} 
```

These are both well-defined and do not risk expansion, so I would argue that something such as `"${PMM_DIR}"/tools` makes things clear.

Reply:
Just to clarify the part about other shells and lack of syntax-highlighting, you may be on a server where you do not have admin permissions to install packages, `bash` is not present everywhere and `less`, `'more`, or `vi` may be all that you have to read the file before executing it.

Reply:
@cezmunsta 
I'm not sure we should sacrifice the convenience of all in favor people don't have syntax-highlighting. Yes, sometimes people don't have syntax-highlighting but it's rare these days.

The main users of our script have highlighting and they should be our `target auditory`. If we can help others, that's great but without decrease to the comfort of others.

In any case, I think it's better to follow the guides and I don't see any strong recommendation about this:
https://google.github.io/styleguide/shellguide.html#quoting

You can choose any way that think will be better.

Reply:
@cezmunsta if most administrators and developers don't even have access to `vim`, then I think IT industry is failing :)

Reply:
@EvgeniyPatlan any thoughts?